libraries/AP_HAL/utility/Socket.h

```diff
@@ -28,6 +28,9 @@
 #include <netinet/tcp.h>
 #include <arpa/inet.h>
 #include <sys/select.h>
+#if defined(__ANDROID__) && defined(__BORLANDC__)
```

Comment:
i solve it including the config output into the IDE after waf build the system. Thanks @lucasdemarchi .